File: Solution/others/Java/SahilK/adv spring/demo/src/main/java/com/transflower/tflassessment/demo/services/ResultService.java
package com.transflower.tflassessment.demo.services;

import java.time.LocalDateTime;
import java.util.List;

import com.transflower.tflassessment.demo.entities.CandidateResultDetails;
import com.transflower.tflassessment.demo.entities.TestResultDetails;

public interface ResultService {

    int getCandidateScore(int candidateId, int testId);

    // public Task<int> GetCandidateTestScore(int candidateId, int testId);
    boolean setCandidateTestStartTime(int candidateId, int testId, LocalDateTime time);

    boolean setCandidateTestEndTime(int candidateId, int testId, LocalDateTime time);

    CandidateResultDetails candidateTestResultDetails(int candidateId, int testId);

    List<TestResultDetails> getTestResultDetail(int testId);
}

File: Solution/WebSolution/Dotnet/AssessmentPortal/Monolithic/Monolithic_Minimal/AssessmentServices/Implementations/ResultService.cs
using MySql.Data.MySqlClient;
using System.Data;
using Transflower.TFLAssessment.Entities;
using Transflower.TFLAssessment.Repositories.Interfaces;
using Transflower.TFLAssessment.Services.Interfaces;
namespace Transflower.TFLAssessment.Services.Implementations;

//Providers
public class ResultService :IResultService
{ 
     private readonly IResultRepository _repository;

    public ResultService(IResultRepository repository)
    {
        _repository = repository;

    }

        public int GetCandidateScore(int candidateId, int testId)
    { 
        return _repository.GetCandidateScore(candidateId,testId);
    }
    public int GetCandidateTestScore(int candidateId, int testId)
    {
        return _repository.GetCandidateTestScore(candidateId,testId);
    }
    public CandidateResultDetails CandidateTestResultDetails(int candidateId, int testId)
    {
        return _repository.CandidateTestResultDetails(candidateId,testId);
    }

}
